File: builder-pattern/python/src/PhoneBuilder.py
# -*- coding: utf-8 -*-
"""
@author: jarry
"""

from src.Builder import Builder
from src.Phone import Phone

class PhoneBuilder(Builder):
  phone = None

  def __init__(self):
    pass

  def reset(self):
    self.phone = Phone()

  def setName(self, name):
    self.phone.setName(name)
    return self

  def setScreen(self, screen):
    self.phone.setScreen(screen)
    return self

  def setGPU(self, no):
    self.phone.setGpuType(no)
    return self

  def getProduct(self):
    return self.phone

File: builder-pattern/ts/src/PhoneBuilder.ts
import { Builder } from './Builder.js'
import { Phone } from './Phone.js'

export class PhoneBuilder implements Builder {
  phone: Phone | any
  constructor() {
    // this.reset()
  }

  reset() {
    this.phone = new Phone()
  }

  setName(name: string): PhoneBuilder {
    this.phone.setName(name)
    return this
  }

  setScreen(screen: []): PhoneBuilder {
    this.phone.setScreen(screen)
    return this
  }

  setGPU(no: number): PhoneBuilder {
    this.phone.setGpuType(no)
    return this
  }

  getProduct() {
    return this.phone
  }
}
